sql/sql_optimizer.cc

```diff
@@ -10151,8 +10151,9 @@ static bool internal_remove_eq_conds(THD *thd, Item *cond,
     *cond_value=Item::COND_UNDEF;
     Item *item;
 
-    bool is_cond_or =
-      (cond && dynamic_cast<Item_cond_or*>(cond)) ? true : false;
+    bool is_cond_or = dynamic_cast<Item_cond_or*>(cond) ? true : false;
+		bool is_cond_and = dynamic_cast<Item_cond_and*>(cond) ? true : false;
```

Comment:
1:  Code alignement. 
2: For example:
const bool and_level= item_cond->functype() == Item_func::COND_AND_FUNC;

Reply:
that's good